### PR TITLE
 Add PRIMER_DIRECTORY_URL to client.

### PR DESCRIPTION
**Pull request by eyurko**
sorry for the extra commit -- shouldn't be there.

### CODE REVIEW

File: client/config/deploymentSpecificConfig.js [hunk 1]
```diff
@@ -17,3 +17,5 @@
  */
 
 tieConfig.constant('SERVER_URL', null);
+
+tieConfig.constant('PRIMER_URL_DIRECTORY', '/docs/');
```
Comment (seanlip) [inline client/config/deploymentSpecificConfig.js]: PRIMER_DIRECTORY_URL, perhaps? (Since it's a URL to the Primer directory.)

File: client/question/components/PrivacyModalDirective.js [hunk 2]
```diff
@@ -111,9 +111,9 @@ tie.directive('privacyModal', [function() {
       </style>
     `,
     controller: ['$scope', '$window', 'CookieStorageService',
-      'ServerHandlerService', 'MENU_PAGE_URL_FROM_QUESTION_PAGE',
+      'ServerHandlerService', 'CLIENT_VERSION_HOMEPAGE_URL',
```
Comment (seanlip) [inline client/question/components/PrivacyModalDirective.js]: Maybe call it the GITHUB_HOMEPAGE_URL or GITHUB_PAGE_URL since that's more unambiguous (I think of "client version URL" as the homepage of the standalone version, i.e. /question or perhaps /menu).

Also maybe this stuff should be a separate PR?

File: client/question/components/TextWithMarkdownLinksSnippetDirective.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ tie.directive('textWithMarkdownLinksSnippet', [function() {
             /\[([^[\]]+)\]\(([^)]+)\)/g,
             function(match, p1, p2) {
               var startsWithHttps = (p2.indexOf('https://') === 0);
-              var goesToPrimer = (p2.indexOf('../docs/py-primer-') === 0);
+              var goesToPrimer = (p2.indexOf('primer-url') === 0);
```
Comment (seanlip) [inline client/question/components/TextWithMarkdownLinksSnippetDirective.js]: This is non-standard, we should have docs/comments explaining the special format of primer URLs.

File: client/question/components/TextWithMarkdownLinksSnippetDirectiveSpec.js [hunk 4]
```diff
@@ -71,37 +73,42 @@ describe('TextWithMarkdownLinksSnippetDirective', function() {
       expect(template.html()).toContain(
         'abc <a href="https://www.google.com" target="_blank">def</a>');
 
-      parentScope.parentContent = 'abc [def](../docs/py-primer-light.html)';
+      parentScope.parentContent = 'abc [def](primer-url)';
       scope.$digest();
       expect(template.html()).toContain(
-        'abc <a href="../docs/py-primer-light.html" target="_blank">def</a>');
+        'abc <a href="..' + PRIMER_URL_DIRECTORY +
```
Comment (seanlip) [inline client/question/components/TextWithMarkdownLinksSnippetDirectiveSpec.js]: Why not include the .. in PRIMER_URL_DIRECTORY?

(and/or make the name clearer to indicate whether it's the directory from the current working directory, or from TIE's basepath)

File: client/question/components/PrivacyModalDirective.js (hunk 2 above)
Comment (eyurko) [reply client/question/components/PrivacyModalDirective.js]: yeah, probably, sorry, I think I must have grabbed it in an extra commit.

File: client/question/components/TextWithMarkdownLinksSnippetDirective.js (hunk 3 above)
Comment (eyurko) [reply client/question/components/TextWithMarkdownLinksSnippetDirective.js]: will do!

File: client/question/components/TextWithMarkdownLinksSnippetDirectiveSpec.js (hunk 4 above)
Comment (eyurko) [reply client/question/components/TextWithMarkdownLinksSnippetDirectiveSpec.js]: makes sense. I'll just include the ..

File: client/config/deploymentSpecificConfig.js (hunk 1 above)
Comment (eyurko) [reply client/config/deploymentSpecificConfig.js]: yeah I prefer that

File: client/question/components/TextWithMarkdownLinksSnippetDirective.js [hunk 5]
```diff
@@ -41,7 +41,11 @@ tie.directive('textWithMarkdownLinksSnippet', [function() {
             /\[([^[\]]+)\]\(([^)]+)\)/g,
             function(match, p1, p2) {
               var startsWithHttps = (p2.indexOf('https://') === 0);
-              var goesToPrimer = (p2.indexOf('../docs/py-primer-') === 0);
+              // The URL for the python primer needs to be special-cased so that
+              // we can dynamically link to the content, as its stored in
```
Comment (seanlip) [inline client/question/components/TextWithMarkdownLinksSnippetDirective.js]: its --> it's (or, better, "the primer docs")

File: client/question/components/TextWithMarkdownLinksSnippetDirective.js [hunk 6]
```diff
@@ -41,7 +41,11 @@ tie.directive('textWithMarkdownLinksSnippet', [function() {
...
+              // different locations depending on whether you're using the open-
+              // source version or a standalone hosted version.
```
Comment (seanlip) [inline client/question/components/TextWithMarkdownLinksSnippetDirective.js]: Drop "standalone"; we often use that term to refer to the GitHub version.
